### DISCUSSION TITLE
Support target WebAssembly and WebGPU

### DISCUSSION THREAD

**Discussion by Solido** [openrndr/openrndr#389] [Ideas]
https://github.com/nannou-org/nannou based on Rust is planning to support this targets so it open compatibility with a wide range of devices from browser to mobiles but also some interoperability with other libs.

Kotlin-native will support it natively when available but it require some studies to see if OpenRNDR can go this way too.

Thanks !

**Comment by hamoid** [comment]
@Solido Hi! :) I'm also interested in running OPENRNDR on the browser / mobile. 
This would make https://github.com/openrndr/openrndr/issues/103 no longer relevant.

Feel free to add news here about Kotlin with WebGPU / WebAssembly.

**Comment by hamoid** [comment]
An example of what I imagine possible: creating something like this (full browser window with animated graphics, loads very fast) https://www.funkykarts.rocks/demo.html with OPENRNDR (more examples at https://www.webassemblygames.com/ )

And progress with kotlin to webassembly: https://discuss.kotlinlang.org/t/state-of-webassembly-in-kotlin/18348

**Comment by Solido** [comment]
http://wasm.continuation-labs.com/d3demo
Port for ID4 game engine

**Comment by hamoid** [comment]
From https://blog.jetbrains.com/kotlin/2021/05/nine-highlights-from-the-kotlin-roadmap/

> 3️⃣ Placing a bet on the WebAssembly
> We believe that WebAssembly will become the new standard for creating rich web applications in the future, and Kotlin should be a perfect fit there.
> That’s why we’ve decided to go all-in for Kotlin/Wasm! We’ve already started creating a dedicated team that will work on it full time. We plan to iterate on performance, work closely with authors of the WebAssembly GC proposal, implement basic Kotlin language features, libraries, and basic Gradle support, and add experimental JavaScript interop.

**Comment by ygdrasil-io** [comment]
Hey there,

I've been working on this WebGPU binding: https://github.com/wgpu4k/wgpu4k. It's not quite there yet in terms of feature parity, but I'm looking for testers and contributors who might be interested in using it early on.

There's already a version supporting JVM, JavaScript, and WASM published in a public repository. My next goal is to integrate Android and native targets, followed by a bug hunt and then publishing on Maven Central.

@edwinRNDR

**Comment by ygdrasil-io** [comment]
Mobile is now a thing on wgpu4k !
![image](https://github.com/user-attachments/assets/e3a0750e-5f6a-4c82-b000-3ffdab27ff2d)

you can try here :
https://github.com/wgpu4k/hello-cube
